nilearn/plotting/html_stat_map.py

```diff
@@ -0,0 +1,97 @@
+import os
+import tempfile
+import base64
+
+import numpy as np
+from nilearn import image, datasets
+
+from .._utils.extmath import fast_abs_percentile
+from .._utils.param_validation import check_threshold
+from .js_plotting_utils import get_html_template, HTMLDocument
+
+
+def _encode_nii(img):
+    fd, temp = tempfile.mkstemp(suffix='.nii')
+    os.close(fd)
+    try:
+        img.to_filename(temp)
+        with open(temp, 'rb') as f:
+            binary = f.read()
+            encoded = base64.b64encode(binary).decode('utf-8')
+            return encoded
+    finally:
+        os.remove(temp)
+
+
+def _decode_nii(encoded):
+    fd, temp = tempfile.mkstemp(suffix='.nii')
+    os.close(fd)
+    try:
+        with open(temp, 'wb') as f:
+            f.write(base64.b64decode(encoded.encode('utf-8')))
+        img = image.load_img(temp)
+        return image.new_img_like(img, img.get_data())
+    finally:
+        os.remove(temp)
```

Comment:
Does the finally get called despite the return?

Reply:
yes

Reply:
Even cooler, the `finally` block is executed even if the program crashes. Perfect for cleaning up!